src/Microsoft.ML.Data/Utilities/LocalEnvironment.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.ML.Runtime.Data
+{
+    using Stopwatch = System.Diagnostics.Stopwatch;
```

Comment:
What other kind of Stopwatch is there? Can we just have a `using System.Diagnostics;` at the top?

Reply:
`System.Diagnostics` is a pretty big namespace with some names that for whatever reason cleave pretty tightly to common terms in ML (e.g., we have 90% of our code bent around processing instance data, and sure enough we get `Process` and `InstanceData`... :P), after several instances of developer confusion, we decided to adopt this practice. My innate tendency would be to keep up our existing practice of, when you want only a stopwatch, just import only that, rather than to relax it just yet. (Or better yet C# could have a less awkward way to import a single class from a namespace.)

Of course we could relax it. That practice dates from the times before we had channels, where literally *every* component instantiated stopwatches to time themselves, and there were lots of namespace collisions, and so lots of cost. Now that we do have channels, few things produce their own stopwatches -- use of System.Diagnostics is quite limited. So the amount of pain that would be generated now by your suggestion would be much less.

---
In reply to: [218094176](https://github.com/dotnet/machinelearning/pull/923#discussion_r218094176) [](ancestors = 218094176)